src/reactjs/__tests__/CalculatorPage.test.js

```diff
@@ -0,0 +1,61 @@
+import { render, fireEvent, screen } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import renderer from 'react-test-renderer';
+import '@testing-library/jest-dom';
+
+import App from '../components/App';
+import CalculatorPage from '../components/CalculatorPage';
+
+describe('it tests the CalculatorPage Component', () => {
+  it('renders correctly', () => {
+    const tree = renderer
+      .create(<CalculatorPage />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('checks if calcpageTitle title is a descendant of calcpageComponent', () => {
+    render(<CalculatorPage />);
+    const calcpageComponent = screen.getByTestId('calc-1');
+    const calcpageTitle = screen.getByTestId('calc-2');
+    expect(calcpageComponent).toContainElement(calcpageTitle);
+  });
+
+  it('shows 8 in the answer cell when the user clicks the 8 button', () => {
+    render(<BrowserRouter><App /></BrowserRouter>);
```

Comment:
- [ ] [OPTIONAL] I could see we need this line to be present in all next tests so I would highly suggest using setup and teardown in jest, which will save us time and code if we extended the testing we don't have to repeat this code :shipit: 
```js
beforeEach( () => {
render(<BrowserRouter><App /></BrowserRouter>);
});
```
[Setup and Teardown](https://jestjs.io/docs/setup-teardown)